pkg/controller/kieapp/defaults/defaults.go

```diff
@@ -159,3 +169,45 @@ func parseTemplate(e v1.EnvTemplate, objBytes []byte) []byte {
 
 	return b.Bytes()
 }
+
+func convertToConfigMapName(filename string) (configMapName, file string) {
+	name := constants.ConfigMapPrefix
+	result := strings.Split(filename, "/")
+	if len(result) > 1 {
+		name = strings.Join([]string{name, result[0]}, "-")
```

Comment:
Am I missing something, or should this be results[len(results) - 1]? I assume you want to handle things like env/trial.yaml, no?

Reply:
this is actually naming the configmap based on the directory the file resides in. that configmap will then contain a map of all the files (in yaml form) that that directory is home to.

Reply:
I'd like to hear the pros and cons of individual configmaps versus grouping them together. I worry that this would be the worst of all worlds. I assume an individual configmap (per yaml file) is easier to create manually, edit, etc... on the other hand, grouping them together will create less clutter and organize everything neatly into one configmap for the operator. Any other pros/cons?

Reply:
one of the main reasons i went this direction was configmaps didn't support special characters in the object name. so, things like '/' and '.yaml' were failing to create. the other reason is, this mirrors how the oc cli creates configmaps. e.g.
`oc create configmap kieconfig-envs --from-file config/envs/`

Reply:
Ok, that's fair... another issue is that in your example above, the code will use "config" for the name, whereas you mean it should be "env".  So I basically understand that the method accepts either a single file, or a file within one directory deep, and that's all we need today. This is not high priority, but ideally we would avoid potential for future bugs, where enhancements require two levels deep directories. I don't think you need to support that now, but should make clear it's not supported. Best if that's done through code, the signature only takes a directory name and/or file name, there are two separate functions, or it just causes an error if file path has more than two parts. It can also just be a comment on top of the function.

Reply:
I've enhanced the code to support all levels of directory depth